### DISCUSSION TITLE
Adding Temper to Miyoku ZMK

### DISCUSSION THREAD

**Discussion by calerouxz** [manna-harbour/miryoku#286] [Q&A] [answered]
I would like to use the [Temper ](https://github.com/raeedcho/temper) with Miryoku, it is not currently supported. It also does not seem to work properly using the Corne configuration as you seemingly could for the Chocofi. I tested this and it would appear as the columns were shifted 1 column to the right and the pinky column not working, after flashing the example firmware from [temper zmk config](https://github.com/raeedcho/temper-zmk-config), all keys work correctly.

Can anyone provide instructions to, or help add this keyboard?

**Comment by manna-harbour** [comment, marked as answer]
There are no instructions yet, but there are plenty of examples in the commit history.

This one https://github.com/manna-harbour/miryoku_zmk/commit/d43e5ddfdcb73e346c8a39173bf63ebbb969a867 is very similar, as they use the same mapping, same MCU, and are both split shields. Just copy the `otg` and  `otg.keymap` files and change `otg` to `temper` in filenames and contents, and replace this line https://github.com/manna-harbour/miryoku_zmk/blob/d43e5ddfdcb73e346c8a39173bf63ebbb969a867/.github/workflows/outboards/shields/otg#L4 in the copy with
```
outboard_repository=raeedcho/temper-zmk-config
```

Otherwise, I'm happy to add support.

**Reply by calerouxz** [reply]
I actually got it to work on my fork, but I've never done a PR before in order to add it to the Miryoku repo. Should I try to add it?

**Reply by manna-harbour** [reply]
Great! Yes, please open a PR. For the PR you'll also need to insert a line here https://github.com/manna-harbour/miryoku_zmk/blob/67a92372ecb8132925e25a8f84d312c2dd048152/.github/workflows/test-all-promicro-shields.yml#L84.

**Reply by manna-harbour** [reply]
Also, for new and modified files, please update the year.

**Reply by calerouxz** [reply]
I've opened the PR, thanks for the help!
